geo/src/algorithm/hausdorff_distance.rs

```diff
@@ -0,0 +1,278 @@
+use crate::GeoFloat;
+use crate::CoordsIter;
+use crate::algorithm::EuclideanDistance;
+use num_traits::Bounded; // used to have T as generic type in folding
+use geo_types::*;
+
+/// Determine the distance between two geometries using the [Hausdorff distance formula].
+///
+/// Hausdorff distance is used to compare two point sets. It measures the maximum euclidean 
+/// distance of a point in one set to the nearest point in another set. Hausdorff distance
+/// is often used to measure the amount of mismatch between two sets. 
+/// 
+/// [Hausdorff distance formula]: https://en.wikipedia.org/wiki/Hausdorff_distance
+pub trait HausdorffDistance<T, Rhs = Self> {
+    fn hausdorff_distance(&self, rhs: &Rhs) -> T;
+}
+
+// We can take advantage of the coords_iter() method for all geometries
+// to iterate across all combos of coords (infinum). Simplifies the 
+// implementations for all geometries.
+macro_rules! impl_hausdorff_distance_coord_iter {
+    ($from:ident, [$($to:ident),*]) => {
+        $(
+            impl<T> HausdorffDistance<T, $to<T>> for $from<T>
+            where
+                T: GeoFloat 
+            {
+                fn hausdorff_distance(&self, geom: &$to<T>) -> T {
+                    // calculate from A -> B 
+                    let hd1 = self
+                        .coords_iter()
+                        .map(|c| {
+                            geom
+                            .coords_iter()
+                            .map(|c2| {
+                                c.euclidean_distance(&c2)
+                            })
+                            .fold(<T as Bounded>::max_value(), |accum, val| accum.min(val))
```

Comment:
Erm, I don't know? Since the map returns `T` not `Option<T>` does `.unwrap_or()` have a use here beyond calling the or portion? Additionally, I was unaware you can call `min()` on an iterator!

Reply:
I've tried replacing this with `.min()` and received the following compiler error. It appears that since `Ord` trait is not implemented for `T` `min()` cannot be used. 
<img width="604" alt="image" src="https://github.com/georust/geo/assets/13534508/2790fbe1-ea9d-4390-8f65-6fb8b3869d9b">